go-controller/pkg/ovn/policy.go

```diff
@@ -1174,6 +1204,23 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) error {
 
 	if len(nsInfo.networkPolicies) == 0 {
 		err = oc.createDefaultDenyPGAndACLs(policy.Namespace, policy.Name, nsInfo)
+		defer func() {
+			if err != nil {
+				nsInfo, nsUnlock, errDelete := oc.ensureNamespaceLocked(policy.Namespace, false, nil)
```

Comment:
There is a potential future race here (whenever we enable parallelization for network policy). Since we gave up nsInfo lock here, its possible another thread processing network policy added another policy while we gave up the lock. In that case we wouldn't want to rollback the default deny stuff. In order to guard against that future problem you can just check  if len(nsInfo.networkPolicies) == 0

Reply:
oh! good point!